Do not search node_modules directories for test files

As title.

Just ran into this via `bl`:

```
> node test_runner.js


Running tests in ./esm/node_modules/.deno/bl@4.1.0/node_modules/bl/test/test.js...

node:internal/modules/cjs/loader:1143
  const err = new Error(message);
              ^

Error: Cannot find module 'tape'
Require stack:
- /Users/joscha/dev/a-node/npm/esm/node_modules/.deno/bl@4.1.0/node_modules/bl/test/test.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/Users/joscha/dev/a-node/npm/esm/node_modules/.deno/bl@4.1.0/node_modules/bl/test/test.js:2:14)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at cjsLoader (node:internal/modules/esm/translators:366:17) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/joscha/dev/a-node/npm/esm/node_modules/.deno/bl@4.1.0/node_modules/bl/test/test.js'
  ]
}

Node.js v20.12.2
error: Uncaught (in promise) Error: npm run test failed with exit code 1
      throw new Error(
            ^
    at runCommand (https://jsr.io/@deno/dnt/0.41.2/lib/utils.ts:56:13)
    at eventLoopTick (ext:core/01_core.js:168:7)
    at async build (https://jsr.io/@deno/dnt/0.41.2/mod.ts:419:5)
    at async file:///Users/joscha/dev/a-node/scripts/build_npm.ts:10:1
```

was able to work around this by using the `rootTestDir` option, e.g.:

```ts
rootTestDir: './src/tests',
```

--- a/lib/utils.test.ts
+++ b/lib/utils.test.ts
@@ -1,6 +1,37 @@
 import * as path from "@std/path";
 import { assertEquals, assertRejects } from "@std/assert";
-import { getDntVersion, runCommand, valueToUrl } from "./utils.ts";
+import { getDntVersion, glob, runCommand, valueToUrl } from "./utils.ts";
+
+Deno.test("glob should not search node_modules or excluded dirs", async () => {
+  const rootDir = await Deno.makeTempDir();
+  try {
+    const testFilePaths = [
+      ["mod.test.ts"],
+      ["sub", "mod.test.ts"],
+      ["node_modules", "pkg", "mod.test.ts"],
+      ["sub", "node_modules", "pkg", "mod.test.ts"],
+      ["npm", "mod.test.ts"],
+    ];
+    for (const filePath of testFilePaths) {
+      const absPath = path.join(rootDir, ...filePath);
+      await Deno.mkdir(path.dirname(absPath), { recursive: true });
+      await Deno.writeTextFile(absPath, "");
+    }
+
+    const paths = await glob({
+      pattern: "**/*.test.ts",
+      rootDir,
+      excludeDirs: [path.join(rootDir, "npm")],
+    });
+
+    assertEquals(
+      paths.map((p) => path.relative(rootDir, p)).sort(),
+      ["mod.test.ts", path.join("sub", "mod.test.ts")],
+    );
+  } finally {
+    await Deno.remove(rootDir, { recursive: true });
+  }
+});
 
 Deno.test({
   name: "should error when command doesn't exist",

--- a/lib/utils.ts
+++ b/lib/utils.ts
@@ -3,7 +3,11 @@
 import { expandGlob } from "@std/fs/expand-glob";
 import * as path from "@std/path";
 
-/** Gets the files found in the provided root dir path based on the glob. */
+/**
+ * Gets the files found in the provided root dir path based on the glob.
+ *
+ * Any `node_modules` directory is never searched.
+ */
 export async function glob(options: {
   pattern: string;
   rootDir: string;
@@ -14,7 +18,7 @@ export async function glob(options: {
     root: options.rootDir,
     extended: true,
     globstar: true,
-    exclude: options.excludeDirs,
+    exclude: [...options.excludeDirs, "**/node_modules"],
   });
   for await (const entry of entries) {
     if (entry.isFile) {

--- a/mod.ts
+++ b/mod.ts
@@ -98,7 +98,11 @@ export interface BuildOptions {
   skipSourceOutput?: boolean;
   /** Root directory to find test files in. Defaults to the cwd. */
   rootTestDir?: string;
-  /** Glob pattern to use to find tests files. Defaults to `deno test`'s pattern. */
+  /**
+   * Glob pattern to use to find tests files. Defaults to `deno test`'s pattern.
+   *
+   * Note that `node_modules` directories are never searched.
+   */
   testPattern?: string;
   /**
    * Specifiers to map from and to.

